src/payments/payments.service.ts

```diff
@@ -0,0 +1,89 @@
+import { BadRequestException, Injectable } from '@nestjs/common';
+import { HttpService } from '@nestjs/axios';
+import { firstValueFrom } from 'rxjs';
+import { ZuAppResponse } from 'src/common/helpers/response';
+import { Verification } from 'src/common/Interfaces/payment.interface';
+import { CoursesRepository } from 'src/database/repository/courses.repository';
+import { Connection } from 'typeorm';
+
+@Injectable()
+export class PaymentService {
+  coursesRepository: CoursesRepository;
+  constructor(
+    private readonly connection: Connection,
+    private readonly httpService: HttpService,
+  ) {
+    this.coursesRepository =
+      this.connection.getCustomRepository(CoursesRepository);
+  }
+
+  /**
+   * It takes a reference id as a parameter, makes a request to the paystack api to verify the
+   * transaction, and returns a response to the client
+   * @param {string} ref - This is the transaction reference id gotten from the paystack API.
+   * @returns A response to the client.
+   */
+  async verify(ref: string): Promise<Verification> {
+    try {
+      let result: any;
+      await firstValueFrom(
+        this.httpService.get(`${process.env.PAYSTACK_VERIFICATION_URL + ref}`, {
+          headers: {
+            authorization: `Bearer ${process.env.PAYSTACK_SECRET_KEY}`,
+            'content-type': 'application/json',
+            'cache-control': 'no-cache',
+          },
+        }),
+      )
+        .then((success) => {
+          result = success;
+        })
+        .catch((error) => {
+          result = error;
+        });
+
+      /* This is a check for internet connectivity issues. */
+      if (!result.response && result.status !== 200)
```

Comment:
if result is made the response object of the axios request.. i think result.response is not a part of the result contents ... 
here are the list of parameters in an axios response object
data - the payload returned from the server
status - the HTTP code returned from the server
statusText - the HTTP status message returned by the server
headers - headers sent by server
config - the original request configuration
request - the request object

... perhaps maybe what you should be looking for is -> result.data

Reply:
@Chimexx can u confirm this b4 i merge this

Reply:
@PhilemonBrain, He's right and it's fixed

Reply:
@Chimexx what do you think about babafemis comment on result.reponse

Reply:
I said he's right, and I've pushed an update